projects/ngrx.io/content/guide/store/architecture.md

```diff
@@ -1 +1,418 @@
+
+So why do we need a Pub-Sub pattern on the above application? The answer is loose coupling. I we don't use a pattern like that we most likely have a service talking directly to a component and that's not usually a problem, at first. When you start adding other components suddenly they might want the same data to show and they might even want to change that data. Without our Pub-Sub pattern it's very easy we make a mistake along the road and one component changes the data and the change doesn't reach the other component and suddenly the components are not in agreement what the data should be. 
+
+Ok, so we managed to find some problem areas like
+- tight coupling, making it hard to add new components
+- corrupted state, we made a mistake and suddenly one or more components are not in agreement what the state should be
+
+## Redux
+As was stated earlier Redux is variant of a Pub Sub pattern and it uses certain constructs to make sure we can listen to data from one place and also change the data in a predictable way. 
+
+### Basic concepts
+For Redux to work properly there are some players in there that needs to exist, those are:
+
+- The store, this is the single source of truth, the one place where your data lives, your state. You read from it and if you want to make changes, this is the data that ultimately gets changed. 
+- Reducers, these are functions, guards if you will that ensures that changes happen to the store in a predictable way. Changes to the store can only happen through a reducer. 
+- Actions, these are messages that carry an intent and sometimes a payload. Typical intents are 'Adding a product', 'Login', 'Remove customer'. These are things you want to do. The payload is what we need to change the data to. In the case of 'Adding a product', the payload would be the product we would want to add. Actions gets *dispatched* to a Reducer that processes the action and the end result of that processing is a change to the state of the store
+
+### Redux flow
+
+The first thing that happens in Redux is the following:
+- a store is created
+- listeners are registered to the Store. Listeners usually only cares about specific parts of the state in the Store. The Store when used in an Application is usually a snapshot of what your application contains at a given point and can looks something like this:
+
+```json
+{
+  "language": "english",
+  "products": [{ "id": 1, "name": "movie" }]
+}
+```
+If this is our state above that means the application is showing all its text in `english` and somewhere we are rendering a list of products. Lets say now the user changes the language in the applciation from `english` from `french`. How would we change this? 
+
+#### Changing our state in the store
+
+This is where we create an action representing our intent and because we want to change to something specific we need to have a payload with the value `french`. An action is nothing more than an object looking like this"
```

Comment:
- Consider changing  `and because we want to change to something specific` to `and because our intent is to switch to a specific language`